classes/client/s3_client.php

```diff
@@ -139,6 +139,17 @@ public function get_fullpath_from_hash($contenthash) {
         return "s3://$this->bucket/$filepath";
     }
 
+    /**
+     * Returns s3 trash fullpath to use with php file functions.
+     *
+     * @param  string $contenthash contenthash used as key in s3.
+     * @return string trash fullpath to s3 object.
+     */
+    public function get_trash_fullpath_from_hash($contenthash) {
+        $filepath = $this->get_filepath_from_hash($contenthash);
+        return "s3://$this->bucket/trash/$filepath";
```

Comment:
looks like there some code duplication in get_fullpath_from_hash and  get_trash_fullpath_from_hash.  Probably the cod e can be updated to eliminate it.

Reply:
These two functions are different. They return different paths - path to file and path to file when it's deleted and moved to trash.